### DISCUSSION TITLE
Clarification regarding what characters may precede tags

### DISCUSSION THREAD

**Discussion by chris48s** [jotaen/xit#51] [File Format]
Fairly succinct one:

```
[ ] This item has a #tag
[ ] This item has a#tag
```

Does the second line have a tag?

**Comment by jotaen** [comment]
Currently, yes.

The reasoning is to allow tags to be recognised in use-cases like these:

```
[ ] Do something (#tag1, #tag2)
[ ] Do other thing #tag1/#tag2
```

Initially, I tried to distinguish that more fine-granularly, and to allow tags to be surrounded/preceded only by punctuation characters. I ended up abandoning that, though, because I couldn’t come up with rules that are reasonably straightforward (e.g. can be expressed via a “simple” regex) and universally applicable (in the sense that they work in all languages, not just in English or other western languages). For reference, [this was the original ticket](https://github.com/jotaen/xit/issues/28) and [this the corresponding spec change](https://github.com/jotaen/xit/pull/31).

The current situation has two downsides: (a) the rules are not 100% clear currently, so your original question is hard to answer by merely looking [at the spec](https://github.com/jotaen/xit/blob/main/Specification.md#tag); and (b), it creates false positives / ambiguities, such as:

- `[ ] Worked on website: https://example.org/foo/#some-anchor` → Recognises `#some-anchor` as tag
- `[ ] Change my master password to be bN144#y6Q!Jc` → Recognises `#y6Q` as tag

So far, this didn’t seem to have caused too much trouble in practice. I’m nevertheless open to improving it.

**Reply by chris48s** [reply]
That makes sense. The URL anchor false positive is particularly annoying, but I don't think there is a nice way to recognise `#tag1/#tag2` but not `https://example.com/#anchor` and keep the spec/parsers simple. Without the ability to escape a hash character (not necessarily desirable), I don't think there is any way to completely eliminate false positives.

> The current situation has two downsides: (a) the rules are not 100% clear currently, so your original question is hard to answer by merely looking at the spec

Yes. This is the reason why I asked the question :) The spec should make it clear. Perhaps include an example to clarify it?
